lib/iris/tests/unit/concatenate/test__CoordMetaData.py

```diff
@@ -0,0 +1,111 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""Unit-tests for :class:`iris._concatenate._CoordMetaData`."""
+
+from __future__ import annotations
+
+import numpy as np
+import pytest
+
+from iris._concatenate import (
+    _CONSTANT,
+    _DECREASING,
+    _INCREASING,
+    _CoordMetaData,
+)
+
+from . import ExpectedItem, create_metadata
+
+
+def check(actual: _CoordMetaData, expected: ExpectedItem) -> None:
+    """Assert actual and expected results."""
+    assert actual.defn == expected.defn
+    assert actual.dims == expected.dims
+    assert actual.points_dtype == expected.points_dtype
+    assert actual.bounds_dtype == expected.bounds_dtype
+    assert actual.kwargs == expected.kwargs
+
+
+@pytest.mark.parametrize("order", [_DECREASING, _INCREASING])
+@pytest.mark.parametrize("circular", [False, True])
+@pytest.mark.parametrize("dtype", [np.int32, np.float32])
+@pytest.mark.parametrize("lazy", [False, True])
+@pytest.mark.parametrize("with_bounds", [False, True])
```

Comment:
Hoping to have a future discussion with @ESadek-MO about the trade-off of readability vs utility when it comes to combining Parametrization and Fixtures.